### ISSUE TITLE
Pylint unreliable when parsing property.setter code

### ISSUE THREAD

**Issue by fdebrabander** [pylint-dev/pylint#8752]
### Bug description

Unfortunately cannot give the actual code triggering this issue and unable to reproduce at this moment with a simplified example code. The following example below is **_pseudo code_** to explain, not actual working code.

```
class Parent():
    def __init__():
        self._my_var = None

    @property
    def my_var(self):
        return self._my_var

    @my_var.setter
    def my_var(self, value):
        self._my_var = value


class SubClass(Parent):
    def __init__():
        self.var_1 = None
        self.var_2 = None
        self.var_3 = None
        self.var_4 = None
        self.var_5 = None
        self.var_6 = None
        self.var_7 = None


my_instance = SubClass()
my_instance.myvar = 1337
```

Observations:
- Pylint does not always report an issue, roughly 20% of the invocations it will incorrectly report too-many-instance-attributes. Likely caused by some race condition when parsing a fairly large code base, pylint is used with jobs=0.
- Suppressing this with `disable=too-many-instance-attributes` will then result in reporting of `useless-supression`, but again not with every invocation of pylint.
- The class SubClass has 7 attributes, when pylint reports too-many-instance-attributes, it claims it has 8/7 attributes
- After changing pylint's design_analysis.py method `visit_classdef()` and adding `print(node.instance_attrs)` you can see it included the "my_var" in the list of instance_attrs when pylint reports too-many-instance-attributes. It should not have included "my_var" in the list of node.instance_attrs.

In the actual code base that triggers this issue, all three code parts (two classes and instances of these classes) are in different python files.

### Command used

```shell
pylint source_dir
```

### Pylint output

```shell
my/source/file.py:60:0 R0902: Too many instance attributes (8/7) (too-many-instance-attributes)
```

### Expected behavior

No issue expected.

### Pylint version

```shell
pylint 2.17.3
astroid 2.15.4
Python 3.10.6
```

### OS / Environment

Ubuntu 22.04
